processor/src/stack/mod.rs

```diff
@@ -66,6 +66,35 @@ impl Stack {
         Ok(self.trace[0][self.step])
     }
 
+    /// Return states from the stack, which includes both the trace state and overflow table.
+    /// If n is not passed in, this returns all states.
+    pub fn get_values(&self, n: Option<usize>) -> Vec<Felt> {
+        let size = n.unwrap_or(usize::MAX);
+        let mut states: Vec<Felt> = Vec::new();
+        let trace_states = self.trace_state().to_vec();
+        let mut i = 0;
+        while i < cmp::min(trace_states.len(), size) {
+            let state = trace_states[i];
+            if state == Felt::ZERO {
+                break;
+            }
+            states.push(state);
+            i += 1;
+        }
+
+        if size > states.len() && !self.overflow.is_empty() {
+            let remaining = size - states.len();
+            let overflow_len = self.overflow.len();
+            i = 0;
+            while i < cmp::min(overflow_len, remaining) {
+                states.push(self.overflow[i]);
+                i += 1;
+            }
+        }
+
+        states
+    }
```

Comment:
I think checking for `Felt::ZERO` may not work correctly in all cases - we need to check against the actual stack depth. Also, I think we can simplify this quite a bit. How about something like this:

```Rust
let n = n.unwrap_or(usize::MAX);
let num_items = cmp::min(n, self.depth());

let num_top_items = cmp::min(STACK_TOP_SIZE, num_items);
let mut result = self.trace_state()[..num_top_items].to_vec();

if num_items > STACK_TOP_SIZE {
    let num_overflow_items = num_items - STACK_TOP_SIZE;
    result.extend_from_slice(&self.overflow[..num_overflow_items]);
}

result
```

Reply:
Oh nice, learned something new rust :)